### PR TITLE
Add type checking job to the CI workflow

### PR DESCRIPTION
**Pull request by HiDeoo**
#### Description

Following a Discord conversation, this PR adds a type checking job to the CI workflow to catch type errors in the codebase even tho we are not bundling the code yet.

### CODE REVIEW

**Comment by HiDeoo** [comment]
The unit test failure is related to the Tailwind package which does not match the coverage thresholds expected. This used to pass before the configuration was invalid.

File: packages/tailwind/vitest.config.ts [hunk 1]
```diff
@@ -4,11 +4,13 @@ export default defineConfig({
...
-			lines: 100,
-			functions: 100,
-			branches: 100,
-			statements: 100,
+			thresholds: {
+				autoUpdate: true,
+				lines: 100,
+				functions: 100,
+				branches: 100,
+				statements: 100,
```
Comment (delucis) [inline packages/tailwind/vitest.config.ts:12]: I think it’s fine to drop these down to match the current state — we can aim to get back to full coverage later.

```suggestion
				lines: 94,
				functions: 100,
				branches: 85,
				statements: 94,
```

File: tsconfig.typecheck.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
+  "exclude": ["**/dist/**", "**/__coverage__/**", "packages/starlight/components.ts"]
```
Comment (delucis) [inline tsconfig.typecheck.json:3]: Could it make sense to exclude these in the default tsconfig instead of adding the custom one?

**Review by delucis — APPROVED** [review]
Thank you for tackling this @HiDeoo — looks great. Will be a nice way to catch details that have been slipping through recently.

Left a couple of small comments, but basically LGTM! 🌟

File: tsconfig.typecheck.json (hunk 2 above)
Comment (HiDeoo) [reply tsconfig.typecheck.json:3]: The issue is that we only want to exclude `packages/starlight/components.ts` during type-checking and not while working on Starlight with for example the Astro extension in Visual Studio Code.
`**/dist/**"` and `**/__coverage__/**` are already part of the default tsconfig but unfortunately, when using `extends` in a `tsconfig.json` file, arrays are not merged.

**Pull request merged**